### PR TITLE
Update charts user guide to 0.8

### LINKED ISSUE #1813: Document new Charts design
New design is a big improvements and potentially a magnet for users contribution IF well documented enough so users know how the pattern works and how to create new charts.

### PR DESCRIPTION
**Pull request by fpliger**
issues: closes #1813

### CODE REVIEW

File: sphinx/source/docs/user_guide/charts.rst [hunk 1]
```diff
@@ -42,49 +48,57 @@ initialize your plot with some specific arguments (for chart customization)::
...
+    from bokeh.plotting import output_file, show
+    output_file('histogram.html')
+    show(hist)
 
-Charts support a long list of arguments that you can pass when instantiating a class, as we have shown before,
-but you can also use chained methods to set those attributes as in the following example::
 
-    hist = Histogram(distributions, bins=50, notebook=True)
-    hist.title("chained_methods, dict_input").ylabel("frequency").legend(True).width(400).height(350).show()
+.. image:: /_images/charts_histogram_cdf.png
```
Comment (damianavila) [inline sphinx/source/docs/user_guide/charts.rst]: why not a live plot here?

File: sphinx/source/docs/user_guide/charts.rst [hunk 2]
```diff
@@ -42,49 +48,57 @@ initialize your plot with some specific arguments (for chart customization)::
...
 * ``height`` (int): the height of you plot in pixels.
-* ``tools`` (bool): to enable or disable the tools in your plot.
-* ``filename`` (str or bool): the name of the file where your plot will be written.
-* ``server`` (str or bool): the name of your plot in the server.
-* ``notebook`` (bool):if you want to output (or not) your plot into the IPython notebook.
+* ``tools`` (str or bool): to enable or disable the tools in your chart.
+* ``palette`` (list): a list containing the colormap as hex values.
+* ``filename`` (str or bool): the name of the file where your chart will be written.
+* ``server`` (str or bool): the name of your chart in the server.
+* ``notebook`` (bool):if you want to output (or not) your chart into the IPython notebook.
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/charts.rst]: I think this call all be automated, but later

**Comment by bryevdv** [comment]
LGTM

File: sphinx/source/docs/user_guide/charts.rst [hunk 3]
```diff
@@ -559,22 +607,52 @@ Interface outputs
...
+.. _charts_builders:
+
+Chart Builders
+--------------
+
+Since 0.8 release `Charts` creation is streamlined by specific
+objects called Builders. Builders are convenience classes that create
+all computation, validation and low-level geometries needed to render a High Level
+Chart. This provides clear pattern to easily extend the `Charts` interface
+with new charts. For more info about this refer to the
```
Comment (damianavila) [inline sphinx/source/docs/user_guide/charts.rst]: refer to the ???

**Comment by damianavila** [comment]
Well, there are several png images that can be changed later for real interactive plots... maybe something for later...

File: sphinx/source/docs/user_guide/charts.rst (hunk 3 above)
Comment (damianavila) [inline sphinx/source/docs/user_guide/charts.rst]: I think you forgot the reference here...

**Comment by damianavila** [comment]
Just one important comment (the missing reference), otherwise, LGTM...

File: sphinx/source/docs/user_guide/charts.rst (hunk 3 above)
Comment (fpliger) [inline sphinx/source/docs/user_guide/charts.rst]: grrr... yes. Ideally I wanted to write a "how to" guide to build new charts.. but won't make in time for the release. Probably best is to refer to the references for now.

Comment (damianavila) [inline sphinx/source/docs/user_guide/charts.rst]: yep

**Pull request merged**